go/vt/wrangler/materializer.go

```diff
@@ -644,45 +680,48 @@ func (mz *materializer) generateInserts(ctx context.Context) (string, error) {
 			rule := &binlogdatapb.Rule{
 				Match: ts.TargetTable,
 			}
-			// Validate the query.
-			stmt, err := sqlparser.Parse(ts.SourceExpression)
-			if err != nil {
-				return "", err
-			}
-			sel, ok := stmt.(*sqlparser.Select)
-			if !ok {
-				return "", fmt.Errorf("unrecognized statement: %s", ts.SourceExpression)
-			}
-			if mz.targetVSchema.Keyspace.Sharded && mz.targetVSchema.Tables[ts.TargetTable].Type != vindexes.TypeReference {
-				cv, err := vindexes.FindBestColVindex(mz.targetVSchema.Tables[ts.TargetTable])
+
+			if ts.SourceExpression != "" {
```

Comment:
Change this to:
```
if ts.SourceExpression == "" {
  continue
}
```

Reply:
but don't i still need to add to the Filter.Rules after this block?

Reply:
Oh. Right. Missed that. You can duplicate that append line here before continuing.
Main concern is that there are already too many indents. I'm not happy about the existing if-else block already.

Reply:
duped. also i pulled up the `filter` as default assign so that we can get rid of an else as well.